### PR TITLE
refactor: add tldr summary to seo description plus test

### PR DESCRIPTION
**Pull request by ayadotdev**
### Describe what this PR does

This PR adds a TLDR summary to the seo description.

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

DD-520 #done

### CODE REVIEW

File: packages/webapp/__tests__/PostPage.tsx [hunk 1]
```diff
@@ -226,6 +226,34 @@ it('should show login on upvote click', async () => {
   expect(showLogin).toBeCalledTimes(1);
 });
 
+it('should check meta tag', async () => {
```
Comment (rebelchris) [inline packages/webapp/__tests__/PostPage.tsx:229]: I would name these a bit more explicit as:
"should check meta tag with only summary"

File: packages/webapp/__tests__/PostPage.tsx [hunk 2]
```diff
@@ -226,6 +226,34 @@ it('should show login on upvote click', async () => {
...
+  const seo = getSeoDescription(
+    createPostMock({
+      summary: 'Test summary',
+    }).result.data.post,
+  );
+
+  expect(seo).toEqual('Test summary');
+});
+
+it('should check meta tag', async () => {
```
Comment (rebelchris) [inline packages/webapp/__tests__/PostPage.tsx:239]: "should check meta tag with only description"

File: packages/webapp/__tests__/PostPage.tsx [hunk 3]
```diff
@@ -226,6 +226,34 @@ it('should show login on upvote click', async () => {
...
+  const seo = getSeoDescription(
+    createPostMock({
+      description: 'Test description',
+    }).result.data.post,
+  );
+
+  expect(seo).toEqual('Test description');
+});
+
+it('should check meta tag', async () => {
```
Comment (rebelchris) [inline packages/webapp/__tests__/PostPage.tsx:249]: "should check meta tag with no description and summary"

File: packages/webapp/__tests__/PostPage.tsx [hunk 4]
```diff
@@ -226,6 +226,34 @@ it('should show login on upvote click', async () => {
...
+  const seo = getSeoDescription(
+    createPostMock({
+      description: 'Test description',
+    }).result.data.post,
+  );
+
+  expect(seo).toEqual('Test description');
+});
+
```
Comment (rebelchris) [inline packages/webapp/__tests__/PostPage.tsx:248]: Also we should probably add one with both summary and description to make sure it favours the summary.

File: packages/webapp/pages/posts/[id].tsx [hunk 5]
```diff
@@ -65,6 +66,15 @@ const ShareNewCommentPopup = dynamic(
 );
 const Custom404 = dynamic(() => import('../404'));
 
+export const getSeoDescription = (post: Post): string => {
+  if (post?.summary?.length > 0) {
+    return post?.summary;
+  }
+  if (post?.description) {
```
Comment (rebelchris) [inline packages/webapp/pages/posts/[id].tsx:73]: Why don't we check for length here?
Could you try with a empty description object and see what happens?

File: packages/webapp/pages/posts/[id].tsx [hunk 6]
```diff
@@ -249,7 +256,7 @@ const PostPage = ({ id, postData }: Props): ReactElement => {
         <Head>
           <link rel="preload" as="image" href={postById.post.image} />
         </Head>
-        <NextSeo {...seo} />
+        <NextSeo data-testId="post-page-seo-id" {...seo} />
```
Comment (rebelchris) [inline packages/webapp/pages/posts/[id].tsx:259]: You can remove the testId here.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice work!
Some small things, but great setup 💖

File: packages/webapp/pages/posts/[id].tsx (hunk 6 above)
Comment (ayadotdev) [reply packages/webapp/pages/posts/[id].tsx]: All seems done, thank you! @rebelchris

**Comment by ayadotdev** [comment]
All seems done, thank you! @rebelchris

**Review by rebelchris — APPROVED** [review]
Looks great! thanks on these changes
Was the length check needed?

**Review by sshanzel — APPROVED** [review]
Great work on this! 🙌

**Review by rebelchris — APPROVED** [review]
Think we are ready to go here 💖

**Pull request merged**